mvt/android/modules/androidqf/sms.py

```diff
@@ -58,11 +58,18 @@ def parse_backup(self, data):
 
         password = None
         if header["encryption"] != "none":
-            password = getpass.getpass(prompt="Backup Password: ", stream=None)
+            if self.interactive:
+                password = getpass.getpass(prompt="Backup Password: ", stream=None)
+            else:
+                self.log.warning(
+                    "Cannot decrypt backup, because interactivity"
+                    " was disabled and the password is unknown"
+                )
```

Comment:
Your PR does a good job at avoiding prompting for a password interactively! If this only happens when parsing Android backups then I think we should also add an option to provide a password non-interactively to `mvt-android`.

I imagine a use case where an analyst has collected 50 AndroidQF archives with a common backup encryption password. The analyst would like to run MVT against all of the archives and analyse the enclosed backups too.

MVT should skip prompting for the password if a password is already set as an argument or env variable.

```
mvt-android check-backup --backup-password "ABCD" backup.ab
mvt-android check-backup -p "ABCD" backup.ab
mvt-android check-androidqf --backup-password "ABCD" androidqf_output.zip
mvt-android check-androidqf  -p "ABCD" androidqf_output.zip
MVT_ANDROID_BACKUP_PASSWORD="ABCD" mvt-android check-androidqf androidqf_output.zip
````
`mvt-ios` already has similar functionality from loading a backup password from an argument or an environment variable. 

https://github.com/mvt-project/mvt/blob/main/mvt/ios/cli.py#L99-L126

Reply:
I was looking into how to implement something like this and I'm unsure how to pass the backup password down to the `mvt/android/modules/androidqf/sms.py` module.

As far as I could find there's no existing way to pass a module context from the outside (except for the context every module gets of course) or did I miss anything?

The way I would implement this is to give modules access to their parent `Command`, which could then backup password in a field and make it available to the `Module`.
This would be a pretty invasive change, since it would have to change the signature of every module to allow for that change.
What do you think? Is that an acceptable trade-off or do you see a better way of implementing this?

Reply:
Good point! We didn't have a good way to pass module specific options down to a command without adding a module parameter which would change the function signature and require changes to every module.

I've just merged a PR #367 which fixed this by creating adding `module_options` parameter. This is a breaking change to any tools which use MVT modules directly but should make things more maintainable going forward, including this PR.

Could you update your PR by passing `non_interactive` and `backup_password` in `module_options` instead? Thanks again for the PR :)